masp_primitives/src/transaction.rs

```diff
@@ -26,7 +26,40 @@ pub mod memo;
 pub mod serialize;
 pub mod util;
 
-pub type GrothProofBytes = [u8; GROTH_PROOF_SIZE];
+// pub type GrothProofBytes = [u8; GROTH_PROOF_SIZE];
+#[derive(PartialEq, Copy, Clone, Debug)]
+pub struct GrothProofBytes(pub [u8; GROTH_PROOF_SIZE]);
+
+impl Default for GrothProofBytes {
+    fn default() -> Self {
+        GrothProofBytes([0u8; GROTH_PROOF_SIZE])
+    }
+}
+
+impl AsRef<[u8]> for GrothProofBytes {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl From<[u8; GROTH_PROOF_SIZE]> for GrothProofBytes {
+    fn from(value: [u8; GROTH_PROOF_SIZE]) -> GrothProofBytes {
+        GrothProofBytes(value)
+    }
+}
+
+impl BorshSerialize for GrothProofBytes {
+    fn serialize<W: Write>(&self, writer: &mut W) -> borsh::maybestd::io::Result<()> {
+        writer.write_all(self.as_ref())?;
+        Ok(())
+    }
+}
+
+impl BorshDeserialize for GrothProofBytes {
+    fn deserialize(buf: &mut &[u8]) -> borsh::maybestd::io::Result<Self> {
+        read_zkproof(buf)
+    }
+}
 
```

Comment:
I think if we use `borsh = {version = "0.9", features = ["const-generics"]}` and `pub type GrothProofBytes` then we get all of these for free, without wrapping `GrothProofBytes` in a struct - unless there is another thing that needs it?

Reply:
Wow, that's great!